src/ocean_data_assim/MOM_oda_driver.F90

```diff
@@ -124,7 +161,7 @@ subroutine init_oda(Time, G, GV, CS)
   type(hor_index_type), pointer :: HI=> NULL()
   type(directories) :: dirs
 
-  type(grid_type), pointer :: T_grid !< global tracer grid
+  type(grid_type), pointer :: T_grid => NULL() !< global tracer grid
```

Comment:
As identified in #1456, we ought to stop initializing our pointers in subroutines like this, since it implicitly tags them with `save`.

I doubt it is an issue here since the function is probably only called once, but it would be safer to initialize this pointer with an explicit statement.  (Same for `HI`, although it's not part of this PR).